use uv + pyproject.toml instead of requirements.txt 💜

soustruh opened the pull request.

soustruh committed:
use uv + pyproject.toml instead of requirements.txt 💜

kudj commented:
@soustruh could you please change app id a vendor https://github.com/keboola/cookiecutter-python-component/blob/main/%7B%7Bcookiecutter.repository_folder_name%7D%7D/.github/workflows/push.yml#L6-L7

either as cookiecutter variable or add # TODO to those lines

also please let's fix the current branch: https://github.com/keboola/cookiecutter-python-component/blob/main/%7B%7Bcookiecutter.repository_folder_name%7D%7D/.github/workflows/push.yml#L43

thanks!

soustruh committed:
relative paths in Dockerfile

soustruh committed:
render Github Actions workflow using cookiecutter 🍪

soustruh committed:
use dynamic version for generated project

soustruh committed:
cookiecutter itself uses uv 💜

soustruh committed:
Merge branch 'main' into uv

soustruh committed:
improved Running section 🚀

soustruh committed:
default branch name is always main 🫡

kudj commented on {{cookiecutter.repository_folder_name}}/Dockerfile:
```diff
@@ -1,19 +1,20 @@
+COPY pyproject.toml .
 
-WORKDIR /code/
+RUN uv pip install --system .
```
nechtěli jsme sem kopírovat i uv.lock a použít uv sync?

kudj reviewed (approved):
nice work

soustruh committed:
coping folders didn't preserve the folder structure

soustruh committed:
obsolete files removed

soustruh committed:
moving template readme up ⬆️

…so we don't have to delete it from each generated project

soustruh committed:
ChatGPT autocorrect

soustruh committed:
updated gitignore files & minor improvements 💅

ruff formatting, organized imports, os.path → pathlib, …

soustruh committed:
include vendor in KDP app name

soustruh committed:
use normalized component name in pyproject.toml

soustruh committed:
lowercase initial commit message 🙃

soustruh committed:
run the pipeline on any branch name except main

pipeline for main runs only when there's any tag as defined by the tags: ["*"] below

for more information see the docs: https://docs.github.com/en/actions/how-tos/write-workflows/choose-when-workflows-run/trigger-a-workflow#using-filters-to-target-specific-branches-or-tags-for-push-events

kudj reviewed (approved):
super!
